lms/djangoapps/course_blocks/transformers/hidden_content.py

```diff
@@ -0,0 +1,99 @@
+"""
+Visibility Transformer implementation.
+"""
+from openedx.core.lib.block_structure.transformer import BlockStructureTransformer, FilteringTransformerMixin
+from xmodule.seq_module import SequenceModule
+from .utils import collect_merged_boolean_field, collect_merged_date_field
+
+
+class HiddenContentTransformer(FilteringTransformerMixin, BlockStructureTransformer):
+    """
+    A transformer that enforces the hide_after_due field on
+    blocks by removing children blocks from the block structure for
```

Comment:
Good question.  So I specifically remove all children of the top-level subsection that is now hidden, but not the subsection itself.  It's true though that it currently assumes the "hide_after_due" field is set only on subsections.  We should be able to generalize this though, if needed.